src/sparseml/pytorch/models/registry.py

```diff
@@ -91,19 +92,28 @@ def create(
         :param load_strict: True to make sure all states are found in and
             loaded in model, False otherwise; default True
         :param ignore_error_tensors: tensors to ignore if there are errors in loading
+        :param return_extras: True to return the epoch of the loaded model
         :param kwargs: any keyword args to supply to the model constructor
         :return: The instantiated model if key is given else a Tuple containing
-            the instantiated model and the loaded key
+            the instantiated model and the loaded key along with the loaded
+            epoch if present in the checkpoint
         """
         key_copy = key
+        checkpoint = torch.load(pretrained_path)
+        extras = []
 
         if key_copy is None:
-            _checkpoint = torch.load(pretrained_path)
-            if "arch_key" in _checkpoint:
-                key_copy = _checkpoint["arch_key"]
+            if "arch_key" in checkpoint:
+                key_copy = checkpoint["arch_key"]
             else:
                 raise ValueError("No `arch_key` found in checkpoint")
 
+        extras.append(key or key_copy)
+
+        if "epoch" in checkpoint:
```

Comment:
Maybe adding an error or warning if the epoch is not in checkpoint? By default, we should be saving epoch inside the checkpoint right?

Reply:
Same applies to "arch_key" btw